dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/services/QaEventListenerQaService.kt

```diff
@@ -132,12 +136,69 @@ class QaService(
                 "Received document with Hash: $documentId on QA message queue with Correlation Id: $correlationId",
             )
             val messageToSend = objectMapper.writeValueAsString(
-                QaCompletedMessage(documentId, QaStatus.Accepted),
+                QaCompletedMessage(
+                    documentId, QaStatus.Accepted, reviewerIdAutomatedQaService, null,
+                ),
             )
             cloudEventMessageHandler.buildCEMessageAndSendToQueue(
                 messageToSend, MessageType.QaCompleted, correlationId, ExchangeName.DataQualityAssured,
                 RoutingKeyNames.document,
             )
         }
     }
+
+    /**
+     * Method to retrieve qa completed message and store the
+     * @param messageAsJsonString the message body as json string
+     * @param correlationId the correlation ID of the current user process
+     * @param type the type of the message
+     */
+    @RabbitListener(
+        bindings = [
+            QueueBinding(
+                value = Queue(
+                    "dataQualityAssuredQaService",
+                    arguments = [
+                        Argument(name = "x-dead-letter-exchange", value = ExchangeName.DeadLetter),
+                        Argument(name = "x-dead-letter-routing-key", value = "deadLetterKey"),
+                        Argument(name = "defaultRequeueRejected", value = "false"),
+                    ],
+                ),
+                exchange = Exchange(ExchangeName.DataQualityAssured, declare = "false"),
+                key = [RoutingKeyNames.data],
+            ),
+        ],
+    )
+    @Transactional
+    fun addDataReviewFromAutomatedQaToReviewHistoryRepository(
+        @Payload messageAsJsonString: String,
+        @Header(MessageHeaderKey.CorrelationId) correlationId: String,
+        @Header(MessageHeaderKey.Type) type: String,
+    ) {
+        messageUtils.validateMessageType(type, MessageType.QaCompleted)
+        val qaCompletedMessage = objectMapper.readValue(messageAsJsonString, QaCompletedMessage::class.java)
+        val dataId = qaCompletedMessage.identifier
+        val validationResult = qaCompletedMessage.validationResult
+        val reviewerId = qaCompletedMessage.reviewerId
+        if (reviewerId != reviewerIdAutomatedQaService) return
```

Comment:
Warum gibt es hier dieses return?

Reply:
sowohl der QA service als auch der automated-qa service senden qaCompletedMessages.
der qaEventListener soll (in meinen Augen) nur auf die qaCompletedMessages vom automated-qa service hören und quasi nicht auf die message die er selber erzeugt.

Reply:
Wenn wir das hier finalisiert haben, sollte die Documentation über die MessageQueues auch aktualisiert werden. Zu finden sind die momentan hier https://jira.d-fine.dev/browse/DALA-395. Wir sollten dann die aktualisierte Übersicht auch im Repo in "documentation" ablegen

Reply:
Wurde angepasst. Doku wird unabhängig vom merge aktualisiert